### ISSUE TITLE
Dependency conflict with cordova-plugin-file and cordova-plugin-file-transfer

### ISSUE THREAD

**Issue by bozzaj** [Microsoft/cordova-plugin-code-push#336]
### Description

With the recent release of cordova 8.0.0, the cordova team has also released updated versions of their plugins. This is causing a dependency conflict with code-push and the deprecated plugin file-transfer.

file-transfer has been deprecated and has a dependency of file@^5.0.0

code-push has a dependency on file-transfer, as well as file@>=4.3.0

When running cordova prepare, code-push dependencies get installed, so file@6.0.0 (the latest) gets installed, and then file-transfer@1.7.0 gets installed, causing a conflict. file-transfer doesn't see file@6.0.0 as a valid version. Since file-transfer has been deprecated, the cordova team does not plan on any new releases of the plugin.

Will code-push be updated to remove the requirement for the deprecated plugin file-transfer? If not, perhaps changing the dependency of cordova-file to ^5.0.0 so it stays in sync with file-transfer.

### Reproduction

1) Have cordova-plugin-code-push as one of your plugins
2) Remove the platforms and plugins subfolders.
3) Run cordova prepare

Note: Cordova (>=7.0.0) installs plugins alphabetically, so code-push will always be installed before a fixed version of the file plugin.

### Additional Information

cordova-plugin-code-push version: 1.10.1
* List of installed plugins:
Only need cordova-plugin-code-push  to generate the issue.
* Cordova version: 7.0.1
* iOS/Android/Windows version: N/A
* Does this reproduce on a debug build or release build? Both
* Does this reproduce on a simulator, or only on a physical device? N/A - Issue with build phase

**Comment by annakocheshkova** [comment]
Hi @bozzaj, thanks for your request. We are aware of this issue and already have it fixed in one of pull requests. If you have a chance, please try to use it via `cordova plugin add https://github.com/annakocheshkova/cordova-plugin-code-push#56377533a758a0d7790c58bff00fe298deef3837` and see if it works. We will wait for your feedback.

**Comment by sambegin** [comment]
@annakocheshkova , We have the same problem here and I'm trying to get the plugin via 
``` cordova plugin add https://github.com/annakocheshkova/cordova-plugin-code-push#56377533a758a0d7790c58bff00fe298deef3837 ``` 

and we get 
Error: Failed to fetch plugin https://github.com/annakocheshkova/cordova-plugin-code-push#56377533a758a0d7790c58bff00fe298deef3837 via registry.
Probably this is either a connection problem, or plugin spec is incorrect.
Check your connection and plugin name/version/URL.
Failed to get absolute path to installed module

Do we have another way to get the latest fix ?

**Comment by sambegin** [comment]
I had the same problem and 
```
cordova plugin add https://github.com/annakocheshkova/cordova-plugin-code-push#56377533a758a0d7790c58bff00fe298deef3837
```
seems not accessible anymore.


Got it from the latest commit in the Microsoft repo instead : 
``` 
cordova plugin add https://github.com/Microsoft/cordova-plugin-code-push.git#2cd37328399891f3b988ec165e74c53e0cf0525b
```

**Comment by annakocheshkova** [comment]
@sambegin the commit you are mentioning does not yet contain the fix. I have just tried to use `commit 56377533a758a0d7790c58bff00fe298deef3837` and it works for me. 
Please check if you have removed this plugin via `cordova plugin remove` before adding it again. Also, make sure to remove it from the `plugins` folder of your project.

**Comment by sambegin** [comment]
Sorry for the deleted comment. 

I see the the generated plugin.xml file from cordova-plugin-code-push plugin doesn't have the cordova-plugin-file as a dependency anymore... making it all work (since the version of cordova-plugin-file is now only driven by the cordova-plugin-file-transfer). 

Do you know how this stays in sync with the git code project ? When I look at the plugin.xml file in the master branch of Microsoft repo or "annakocheshkova" repo we are still seeing the cordova-plugin-file dependency.

**Comment by sambegin** [comment]
For those who would have had this problem also : 

```
Error: Failed to fetch plugin https://github.com/annakocheshkova/cordova-plugin-code-push#56377533a758a0d7790c58bff00fe298deef3837 via registry.
Probably this is either a connection problem, or plugin spec is incorrect.
Check your connection and plugin name/version/URL.
Failed to get absolute path to installed module
```

you can add the ".git" in the url : 
```
cordova plugin add https://github.com/annakocheshkova/cordova-plugin-code-push.git#56377533a758a0d7790c58bff00fe298deef3837 
```

**Comment by annakocheshkova** [comment]
@sambegin the thing is, our possible fix in the `plugin.xml` is in one of the branches on "annakocheskova" repository, so it does not appear on master. It can only be visible in [this](https://github.com/annakocheshkova/cordova-plugin-code-push/tree/dependency_fix) branch.

**Comment by sambegin** [comment]
@annakocheshkova alright thank you very much ! Seems to work with a fresh cordova build from config.xml file (without platforms / plugins folder) 

Do we know when we expect to have this fix released ?

**Comment by annakocheshkova** [comment]
@sambegin We are planning to release this fix along with the others on the next week.

**Comment by ruslan-bikkinin** [comment]
Hi @bozzaj, we had released new cordova-plugin-code-push version v1.11.0 including fix for this issue. Please upgrade it for your project. 
I'm closing this for now, feel free to re-open if needed.
